src/main/java/frc/robot/subsystems/PhotonSubsystem.java

```diff
@@ -0,0 +1,208 @@
+import edu.wpi.first.networktables.PubSubOption;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ScheduleCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.commands.PhotonMoveToTarget;
+
+//class
+public class PhotonSubsystem extends SubsystemBase {
+
+  //constants
+  private double CAMERA_HEIGHT = 0.29;
+  private Rotation2d CAMERA_PITCH = Rotation2d.fromDegrees(26);
+  //x is forwards, y is sideways with +y being left, rotation probobly if + left too
+  private Pose2d cameraOffset = new Pose2d(new Translation2d(-0.2,0.28), Rotation2d.fromDegrees(0));
+  
+  //networkTableName
+  private String networkTableName = "PhotonCamera";
+  //data max
+  private int maxNumSamples = 10;
+  //declarations
+  private static PhotonSubsystem instance;
+  private DoubleArrayPublisher pubSetPoint;
+  private DoublePublisher pubRange, pubYaw;
+  private PhotonCamera camera1;
+  private Translation2d targetPos;
+  private Rotation2d targetRotation;
+  private LinearFilter filteryaw = LinearFilter.movingAverage(maxNumSamples);
+  private LinearFilter filterX = LinearFilter.movingAverage(maxNumSamples);
+  private LinearFilter filterY = LinearFilter.movingAverage(maxNumSamples);
+  private int numSamples;
+  private int id;
+
+  
+
+  public static PhotonSubsystem getInstance(){
+    if (instance == null){
+      instance = new PhotonSubsystem();
+    }
+    return instance;
+  }
+
+  /** Creates a new photonAprilTag. */
+  private PhotonSubsystem() {
+    //name of camera, change if using multiple cameras
+    camera1 = new PhotonCamera("OV9281");
+    //networktable publishers
+    pubSetPoint = NetworkTableInstance.getDefault().getTable(networkTableName).getDoubleArrayTopic("PhotonAprilPoint").publish(PubSubOption.periodic(0.02));
+    pubRange = NetworkTableInstance.getDefault().getTable(networkTableName).getDoubleTopic("Range").publish(PubSubOption.periodic(0.02));
+    pubYaw = NetworkTableInstance.getDefault().getTable(networkTableName).getDoubleTopic("Yaw").publish(PubSubOption.periodic(0.02));
+    reset(-1);
+  }
+
+  public void reset(int desiredId) {
+    filterX.reset();
+    filterY.reset();
+    filteryaw.reset();
+    //set target info to the robot's info
+    targetRotation = SwerveSubsystem.getInstance().getPose().getRotation();
+    targetPos = SwerveSubsystem.getInstance().getPose().getTranslation();
+    //initialize vars
+    numSamples = 0;
+    id = desiredId;
+  }
+
+  public Command getResetCommand(int tagid){
+    return runOnce(() -> reset(tagid));
+  }
+
+  public Command getWaitForDataCommand(int tagid){
+    return new FunctionalCommand(() -> reset(tagid), ()->{}, (interrupted) ->{}, ()->hasData(), this);
+  }
+  public Command getAprilTagCommand(PhotonPositions spacePositions){
```

Comment:
What you have now works, but relies on the button releasing to cancel this command which isn't ideal.

Here's my second attempt at writing it all inside this command with fancy command requirements:
```Java
  public Command getAprilTagCommand(PhotonPositions spacePositions){
    Command swerveRequirementCommand = Commands.run(() -> {}, SwerveSubsystem.getInstance())
            .withName("AprilTagCommandSwerveRequirement");

    return Commands.sequence(
      getWaitForDataCommand(spacePositions.id),
      Commands.runOnce(() -> swerveRequirementCommand.schedule()), // Add delayed requirement to SwerveSubsystem
      new ProxyCommand( // Proxy these commands to prevent SwerveSubsystem requirement conflicting with swerveRequirementCommand
        Commands.sequence(
          new PhotonMoveToTarget(spacePositions.waypoint,0.1),
          new PhotonMoveToTarget(spacePositions.destination, spacePositions.direction, 0.02 )
      ))
    ).finallyDo(() -> swerveRequirementCommand.cancel()); // Ensure requirement to SwerveSubsystem ends with this command ending
  }
```